adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -14,224 +14,148 @@
+     NSStringFromClass([UnityAds class])];
     if (_adLoadCompletionHandler) {
-      NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-          GADMAdapterUnityErrorDeviceNotSupported, description);
+      NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(GADMAdapterUnityErrorDeviceNotSupported, description);
       _adLoadCompletionHandler(nil, error);
       _adLoadCompletionHandler = nil;
     }
     return;
   }
-
-  [_metaData setCategory:@"mediation_adapter"];
-  [_metaData set:_uuid value:@"load-rewarded"];
-  [_metaData set:_uuid value:_placementID];
-  [_metaData commit];
-  [[GADMAdapterUnitySingleton sharedInstance] requestRewardedAdWithDelegate:weakSelf];
+  
+  [UnityAds addDelegate:self];
+  [UnityAds load:_placementID loadDelegate:self];
 }
 
 - (void)presentFromViewController:(nonnull UIViewController *)viewController {
   if (![UnityAds isReady:_placementID]) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorShowAdNotReady, @"Failed to show Unity Ads rewarded video.");
+    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(GADMAdapterUnityErrorShowAdNotReady, @"Failed to show Unity Ads rewarded video.");
     [_adEventDelegate didFailToPresentWithError:error];
-
-    [_metaData setCategory:@"mediation_adapter"];
-    [_metaData set:_uuid value:@"fail-to-show-rewarded"];
-    [_metaData set:_uuid value:_placementID];
-    [_metaData commit];
     return;
   }
   [_adEventDelegate willPresentFullScreenView];
-
-  [_metaData setCategory:@"mediation_adapter"];
-  [_metaData set:_uuid value:@"show-rewarded"];
-  [_metaData set:_uuid value:_placementID];
-  [_metaData commit];
-
-  [[GADMAdapterUnitySingleton sharedInstance] presentRewardedAdForViewController:viewController
-                                                                        delegate:self];
-}
-
-#pragma mark GADMAdapterUnityDataProvider Methods
-
-- (NSString *)getGameID {
-  return _gameID;
-}
-
-- (NSString *)getPlacementID {
-  return _placementID;
-}
-
-- (void)didFailToLoadWithError:(nonnull NSError *)error {
-  _adLoadCompletionHandler(nil, error);
+  [UnityAds show:viewController placementId:_placementID];
 }
 
 #pragma mark - Unity Delegate Methods
 
 - (void)unityAdsDidError:(UnityAdsError)error withMessage:(nonnull NSString *)message {
-  if (error == kUnityAdsErrorNotInitialized) {
-    if (_adLoadCompletionHandler) {
-      NSError *errorWithDescription =
-          GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
-      _adLoadCompletionHandler(nil, errorWithDescription);
-      _adLoadCompletionHandler = nil;
-    }
-  } else {
-    if (_adEventDelegate) {
-      NSError *errorWithDescription =
-          GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
-      [_adEventDelegate didFailToPresentWithError:errorWithDescription];
-    }
+  if (_adEventDelegate) {
+    [UnityAds removeDelegate:self];
```

Comment:
Move out of `if()` block so it always gets called regardless of `_adEventDelegate` value.